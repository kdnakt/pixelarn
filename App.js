/**
 * @format
 * @flow
 */

import React, {
  Component
} from 'react';
import {
  StyleSheet,
  View
} from 'react-native';
import Pixela from './src/Pixela'

const svgurl = "https://pixe.la/v1/users/kdnakt/graphs/test-graph"

export default class App extends Component {
  constructor(props) {
    super(props)
    this.state = { svgXmlData: null}
  }

  componentDidMount() {
    fetch(svgurl).then(res => this.setState({svgXmlData:res._bodyText}))
  }

  render() {
    return (
      <View style={styles.container}>
        <Pixela data={this.state.svgXmlData} />
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  }
});
