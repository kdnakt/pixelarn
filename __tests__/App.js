import 'react-native'
import React from 'react'
import App from '../App'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<App/>)
  expect(tree.toJSON()).toMatchSnapshot()
})

global.fetch = () => ({
  then: () => {}
})

jest.mock('react-native-svg', () => ({
}));
jest.mock('NativeModules', () => {
  return {
    UIManager: {
      RCTView: {}
    },
    RNGestureHandlerModule: {
      State: {},
      attachGestureHandler: jest.fn(),
      createGestureHandler: jest.fn()
    },
    KeyboardObserver: {}
  } 
})