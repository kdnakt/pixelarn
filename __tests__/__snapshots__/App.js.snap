// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading..
  </Text>
</View>
`;
